### PR TITLE
bb: add page

### LINKED ISSUE #7768: page request: bb
https://babashka.org - shell scripting & task running with [Clojure]

examples: https://book.babashka.org/#_running_a_script

[Clojure]: https://clojure.org

### PR DESCRIPTION
**Pull request by skykanin**
Fixes  #7768

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** v1.0.165

### CODE REVIEW

File: pages/common/bb.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# bb
+
+> Scripting interpreter.
```
Comment (kbdharun) [inline pages/common/bb.md:3]:
```suggestion
```
I don't think this line is necessary as you are already mentioning it below.

File: pages/common/bb.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+> Native Clojure interpreter for scripting.
+> <https://book.babashka.org/#usage>.
```
Comment (kbdharun) [inline pages/common/bb.md:5]:
```suggestion
> More information: <https://book.babashka.org/#usage>.
```

File: pages/common/bb.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+- [e]valuate expression:
```
Comment (kbdharun) [inline pages/common/bb.md:7]:
```suggestion
- [e]valuate an expression:
```
Would this be better?

File: pages/common/bb.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+`bb -e "(+ 1 2 3)"`
+
+- Evaluate a [f]ile:
```
Comment (kbdharun) [inline pages/common/bb.md:11]:
```suggestion
- Evaluate a script [f]ile:
```

File: pages/common/bb.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+`bb -f script.clj`
```
Comment (kbdharun) [inline pages/common/bb.md:13]:
```suggestion
`bb -f {{path/to/script.clj}}`
```
This is the format we use for paths.

File: pages/common/bb.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+- Bind *input* to a lazy seq of lines from stdin:
```
Comment (kbdharun) [inline pages/common/bb.md:15]:
```suggestion
- Bind input to a lazy seq of lines from stdin:
```
Italics wouldn't work properly with some of our clients.

File: pages/common/bb.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+`printf "first\nsecond" | bb -i "(map clojure.string/capitalize *input*)"`
+
+- Bind *input* to a lazy seq of EDN values from stdin:
```
Comment (kbdharun) [inline pages/common/bb.md:19]:
```suggestion
- Bind input to a lazy seq of EDN values from stdin:
```

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page.

File: pages/common/bb.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+- [e]valuate an expression:
+
+`bb -e "(+ 1 2 3)"`
+
+- Evaluate a script [f]ile:
+
+`bb -f {{path/to/script.clj}}`
+
+- Bind input to a lazy seq of lines from stdin:
```
Comment (kbdharun) [inline pages/common/bb.md:14]:
```suggestion
- Bind input to a sequence of lines from stdin:
```
Is the word lazy necessary?

File: pages/common/bb.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+`printf "first\nsecond" | bb -i "(map clojure.string/capitalize *input*)"`
+
+- Bind input to a lazy seq of EDN values from stdin:
```
Comment (kbdharun) [inline pages/common/bb.md:18]:
```suggestion
- Bind input to a sequence of EDN values from stdin:
```

File: pages/common/bb.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+- Evaluate a script [f]ile:
+
+`bb -f {{path/to/script.clj}}`
+
+- Bind input to a sequence of lines from stdin:
+
+`printf "first\nsecond" | bb -i "(map clojure.string/capitalize *input*)"`
+
+- Bind input to a sequence of EDN values from stdin:
```
Comment (blueskyson) [inline pages/common/bb.md:18]: What is END values?

Comment (skykanin) [reply pages/common/bb.md:18]: extensible data notation. It's the data format used to write clojure: https://github.com/edn-format/edn

Comment (kbdharun) [reply pages/common/bb.md:18]: > extensible data notation. It's the data format used to write clojure: https://github.com/edn-format/edn

How about adding it too in the command description?

```suggestion
- Bind input to a sequence of EDN(Extensible Data Notation) values from stdin:
```

**Review by kbdharun — APPROVED** [review]

**Review by blueskyson — APPROVED** [review]

**Pull request merged**